src/datomish/sqlite_schema.cljc

```diff
@@ -18,21 +18,70 @@
 (def current-version 1)
 
 (def v1-statements
-  ["CREATE TABLE datoms (e INTEGER NOT NULL, a SMALLINT NOT NULL, v BLOB NOT NULL, tx INTEGER NOT NULL, index_avet TINYINT NOT NULL DEFAULT 0, index_vaet TINYINT NOT NULL DEFAULT 0)"
+  ["CREATE TABLE datoms (e INTEGER NOT NULL, a SMALLINT NOT NULL, v BLOB NOT NULL, tx INTEGER NOT NULL,
+                         index_avet TINYINT NOT NULL DEFAULT 0, index_vaet TINYINT NOT NULL DEFAULT 0,
+                         index_fulltext TINYINT NOT NULL DEFAULT 0,
+                         unique_value TINYINT NOT NULL DEFAULT 0, unique_identity TINYINT NOT NULL DEFAULT 0)"
    "CREATE INDEX eavt ON datoms (e, a)" ;; No v -- that's an opt-in index.
    "CREATE INDEX aevt ON datoms (a, e)" ;; No v -- that's an opt-in index.
-   "CREATE INDEX avet ON datoms (a, v, e) WHERE index_avet = 1" ;; Opt-in index: only if a has :db/index true.
-   "CREATE INDEX vaet ON datoms (v, a, e) WHERE index_vaet = 1" ;; Opt-in index: only if a has :db/valueType :db.type/ref
+   "CREATE UNIQUE INDEX avet ON datoms (a, v, e) WHERE index_avet = 1" ;; Opt-in index: only if a has :db/index true.
+   "CREATE UNIQUE INDEX vaet ON datoms (v, a, e) WHERE index_vaet = 1" ;; Opt-in index: only if a has :db/valueType :db.type/ref
+
+   ;; TODO: possibly remove this index.  :db.unique/value should be asserted by the transactor in
+   ;; all cases, but the index may speed up some of SQLite's query planning.  For now, it services
+   ;; to validate the transactor implementation.
+   "CREATE UNIQUE INDEX unique_value ON datoms (v) WHERE unique_value = 1"
+   ;; TODO: possibly remove this index.  :db.unique/identity should be asserted by the transactor in
+   ;; all cases, but the index may speed up some of SQLite's query planning.  For now, it serves to
+   ;; validate the transactor implementation.
+   "CREATE UNIQUE INDEX unique_identity ON datoms (a, v) WHERE unique_identity = 1"
+
+   "CREATE INDEX fulltext ON datoms (v) WHERE index_fulltext = 1"
```

Comment:
The situation in which they will not be the same thing: `v` is a row ID that collides with an integer value for a non-fulltext but value-indexed predicate.

We might expect the predicate to be known, in which case it goes back to being equivalent.

And if not — as in the case of a plain ol' fulltext search yielding `rowid`s — yes, we can walk `vaet` and filter for fulltextness.

But given that `index_fulltext` and `index_vaet` (value indexing) are mutually exclusive, it might be fine to just make them separate index chunks. In that case, make this `ON datoms (v, a, e)`.